src/apps/logging/logging.cpp

```diff
@@ -312,27 +274,62 @@ namespace loggingapp
           args.rpc_ctx->set_response_header(
             http::headers::CONTENT_TYPE, http::headervalues::contenttype::TEXT);
           args.rpc_ctx->set_response_body(fmt::format(
-            "Unable to get historical table '{}'.", records.get_name()));
+            "Unable to get table '{}' at {}.{}",
+            records.get_name(),
+            historical_view,
+            historical_seqno));
           return;
         }
 
+        const auto in = params.get<LoggingGetHistorical::In>();
+
         kv::Tx historical_tx;
         auto view = historical_tx.get_view(*historical_map);
-        auto v = view->get(in.id);
+        const auto v = view->get(in.id);
 
         if (v.has_value())
         {
-          args.rpc_ctx->set_response_status(HTTP_STATUS_OK);
-          args.rpc_ctx->set_response_header(
-            http::headers::CONTENT_TYPE, http::headervalues::contenttype::TEXT);
-          args.rpc_ctx->set_response_body(std::move(v.value()));
+          LoggingGetHistorical::Out out;
+          out.msg = v.value();
+          nlohmann::json j = out;
+          ccf::jsonhandler::set_response(std::move(j), args.rpc_ctx, pack);
         }
         else
         {
           args.rpc_ctx->set_response_status(HTTP_STATUS_NO_CONTENT);
         }
       };
 
+      auto is_tx_committed = [this](
```

Comment:
Why is this defined by the app? I understand that this might be nice for the app to override this check but the default seems very much generic to all apps.

Reply:
Yeah this is a bit of a bodge, I'll clean it up. The problem is that we need to capture the `consensus`, at the point of execution, and check it before we request a historical store. Although every historical query handler may want to use this, it still needs to be 'functionalised' to pass it to the functional adapters. But we can pass a bound member function, doesn't need to be a lambda.

Reply:
If I understand this correctly, we could get away without this if we had access to the `consensus` from the historical query adapter - I don't think we need to capture it "at the point of execution" as `consensus` is a long-lived object? I reckon we could pass `consensus` down the the adapter from `frontend.h`?

Reply:
`consensus` is quite long-lived, but not the full node lifetime - we can submit RPCs before we have a consensus (I believe, since we currently have a lot of RPC handlers that expect it might not exist yet). We could pass it to every handler from `frontend.h` in the `Args`, that's probably nicer!

Reply:
So I'd like to do the last idea, passing `consensus` to the handlers that want it rather than having it as a member of the registry. Its a slightly involved change though, and dovetails with making `non-Tx` handlers, so I'd like to do it in a separate PR. Are we happy with the ugly lambda as a temporary solution?

Reply:
Absolutely 👍